### PR TITLE
refactor: get ready to support multiple topics in one wolff_producers

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: src/wolff_producers.erl [hunk 1]
```diff
@@ -376,28 +370,43 @@ start_link_producers(ClientId, Topic, Connections, Config) ->
...
   end;
-maybe_init_producers(St) ->
-  St.
-
-maybe_restart_producers(#{producers_status := ?not_initialized} = St) -> St;
-maybe_restart_producers(#{client_id := ClientId,
-                          topic := Topic,
-                          config := Config
-                         } = St) ->
+init_producers(St, [{Topic, ?initialized} | More], OK, ERR) ->
```
Comment (thalesmg) [inline src/wolff_producers.erl:398]: Missing a clause for `?not_initialized(_Any)`?

File: src/wolff_producers.erl [hunk 2]
```diff
@@ -529,3 +540,31 @@ ensure_timer_cancelled(Tref) when is_reference(Tref) ->
   ok;
 ensure_timer_cancelled(_) ->
   ok.
+
+supervisor_child_id(St) ->
```
Comment (thalesmg) [inline src/wolff_producers.erl:544]: Should the supervisor use this same function to make it consistent by construction?

**Review by thalesmg — APPROVED** [review]

File: src/wolff_producers.erl (hunk 2 above)
Comment (zmstone) [reply src/wolff_producers.erl:544]: producer id is not exactly the supervisor child id.
i'll make it simpler

File: changelog.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
 * 3.0.0
   - Deleted round_robin partition strategy.
-  - Change 'alias' to 'group'.
+  - Change `alias` to `group`.
+    Add `#{group => <<"group1">>}` to producer config for namespacing the topic,
+    so multiple producers for one topic will not clash each other when sharing the same client.
+    The group name should however not clash the client ID --- there is though no such check in the code
+    under the assumption that the application code either uses `group` for all or none producers.
```
Comment (thalesmg) [inline changelog.md:7]:
```suggestion
    under the assumption that the application code either uses `group` for all or no producers.
```

**Review by thalesmg — APPROVED** [review]

**Pull request merged**